cadcdata/cadcdata/core.py

```diff
@@ -320,12 +321,19 @@ def __next__(self):
 
             def next(self):
                 # reads the next raw block
-                data = self._read(self.block_size,
-                                  decode_content=self.decode_content)
+                # Hack warning:
+                # Not using decode_content argument of the Response.read
+                # method in order to get access to the raw bytes and do the
+                # md5 checksum before decoding (decompressing) them.
+                # Unfortunately, that forces us to use the hidden _decode
+                # method of the urllib3 Response class.
+                data = self._read(self.block_size)
                 if len(data) > 0:
                     if md5_check and\
                             (response.headers.get('content-MD5', 0) != 0):
                         hash_md5.update(data)
+                    if self.decode_content:
+                        data = self._decode(data, True, False)
```

Comment:
Is the `_decode()` function private?  Does Python let you do this call?  Oi...

Reply:
Content-MD5 is being deprecated but we (and others) still use it. So if available, we'll take advantage of it.
Nothing is private in Python. _* is just a convention saying "This can change any time. Use it at your own risk"